code/game/objects/items/weapons/handcuffs.dm

```diff
@@ -64,48 +67,145 @@
 
 				log_attack("<font color='red'>[user.name] ([user.ckey]) Attempted to handcuff [C.name] ([C.ckey])</font>")
 
-				var/obj/effect/equip_e/human/O = new /obj/effect/equip_e/human(  )
-				O.source = user
-				O.target = C
-				O.item = user.get_active_hand()
-				O.s_loc = user.loc
-				O.t_loc = C.loc
-				O.place = "handcuff"
-				C.requests += O
-				spawn( 0 )
-					if(istype(src, /obj/item/weapon/handcuffs/cable))
-						feedback_add_details("handcuffs","C")
-						playsound(get_turf(src), 'sound/weapons/cablecuff.ogg', 30, 1, -2)
-					else
-						feedback_add_details("handcuffs","H")
-						playsound(get_turf(src), 'sound/weapons/handcuffs.ogg', 30, 1, -2)
-					O.process()
+				handcuffs_apply(C, user)
 			return
 		else
 			if(!C.handcuffed)
-				var/obj/effect/equip_e/monkey/O = new /obj/effect/equip_e/monkey(  )
-				O.source = user
-				O.target = C
-				O.item = user.get_active_hand()
-				O.s_loc = user.loc
-				O.t_loc = C.loc
-				O.place = "handcuff"
-				C.requests += O
-				spawn( 0 )
-					if(istype(src, /obj/item/weapon/handcuffs/cable))
-						playsound(get_turf(src), 'sound/weapons/cablecuff.ogg', 30, 1, -2)
-					else
-						playsound(get_turf(src), 'sound/weapons/handcuffs.ogg', 30, 1, -2)
-					O.process()
+				handcuffs_apply(C, user)
 			return
+
+/obj/item/weapon/handcuffs/proc/detonate(var/countdown = 1)
```

Comment:
how is if(mode == SYNDICUFFS_ON_APPLY && !charge_detonated) any different from a type check?